### PR TITLE
Jb48660

### PR DESCRIPTION
**Pull request by jlehtoranta**
Only the last 2 commits belong to this PR

### CODE REVIEW

File: src/plugins/mer/merdeployconfiguration.cpp [hunk 1]
```diff
@@ -178,11 +177,16 @@ void MerAddRemoveSpecialDeployStepsProjectListener::updateProject(QmakeProject *
...
 }
 
-bool MerAddRemoveSpecialDeployStepsProjectListener::isAmbienceProject(QmakeProject *project)
+bool MerAddRemoveSpecialDeployStepsProjectListener::isAmbienceProject(Project *project)
 {
-    QmakeProFile *root = project->rootProFile();
-    return root->projectType() == ProjectType::AuxTemplate &&
-        root->variableValue(Variable::Config).contains(QLatin1String(SAILFISH_AMBIENCE_CONFIG));
+    if (qobject_cast<QmakeProject*>(project)) {
+        QmakeProFile *root = qobject_cast<QmakeProject*>(project)->rootProFile();
```
Comment (martyone) [inline src/plugins/mer/merdeployconfiguration.cpp]: Possibly
```
if (auto *qmakeProject = qobject_cast<QmakeProject*>(project)) {
    QmakeProFile *root = qmakeProject->rootProFile();
```

File: src/plugins/mer/merdeploysteps.cpp [hunk 2]
```diff
@@ -223,9 +225,17 @@ bool MerProcessStep::init()
...
+    if (qobject_cast<CMakeBuildConfiguration*>(buildConfiguration())) {
+        bc = qobject_cast<CMakeBuildConfiguration*>(buildConfiguration());
+        if (!bc)
+            bc =  qobject_cast<CMakeBuildConfiguration*>(target()->activeBuildConfiguration());
+    } else {
+        // Default to QmakeBuildConfiguration
+        bc = qobject_cast<QmakeBuildConfiguration*>(buildConfiguration());
+        if (!bc)
+            bc =  qobject_cast<QmakeBuildConfiguration*>(target()->activeBuildConfiguration());
+    }
```
Comment (martyone) [inline src/plugins/mer/merdeploysteps.cpp]: This line is unreachable I think:

`bc =  qobject_cast<CMakeBuildConfiguration*>(target()->activeBuildConfiguration());`

It would be also wort to think if resorting to `target()->activeBuildConfiguration()` is really needed here as `buildConfiguration` already returns that as a fallback, it just doesn't do the casting. Maybe this changed at some point and we can simplify it now here.

File: src/plugins/mer/merdeployconfiguration.cpp [hunk 3]
```diff
@@ -69,7 +69,7 @@ QString MerRpmDeployConfigurationFactory::displayName()
 
 Core::Id MerRpmDeployConfigurationFactory::configurationId()
 {
-    return Core::Id("QmakeProjectManager.MerRpmDeployConfiguration");
+    return Core::Id("Mer.MerRpmDeployConfiguration");
```
Comment (martyone) [inline src/plugins/mer/merdeployconfiguration.cpp]: I would consider moving this renames into a separate commit.

File: src/plugins/mer/merdeployconfiguration.cpp [hunk 4]
```diff
@@ -69,7 +69,7 @@ QString MerRpmDeployConfigurationFactory::displayName()
```
Comment (vigejolla) [inline src/plugins/mer/merdeployconfiguration.cpp]: Remember to update the copyright year :)

File: src/plugins/mer/merdeployconfiguration.cpp (hunk 1 above)
Comment (vigejolla) [reply src/plugins/mer/merdeployconfiguration.cpp]: I would not recommend assignment inside if condition. 

```
auto *qmakeProject = qobject_cast<QmakeProject*>(project);
if (qmakeProject) ...
```

should be preferred.
